0029-stream/0029-stream.md

```diff
@@ -170,19 +172,28 @@ Incoming packets can either be associated with an existing connection, or, for s
+In the most advanced scenarios where clients and servers are both capable of simultaneously sending and receiving, the `ConnectionNewAddress` frame enables bi-directional payment flows. However, the frame is OPTIONAL because clients are commonly not capable of receiving. For example, a client may not be accessible from a routable ILP Address.
 
-Implementations SHOULD wait for a valid response (encrypted with the same shared secret) from the new address to validate the new path. STREAM uses the authenticated request/response packets in lieu of [QUIC's explicit Path Validation](https://quicwg.github.io/base-drafts/draft-ietf-quic-transport.html#rfc.section.6.7). Implementations SHOULD refrain from sending large numbers of packets or large amounts of data to a new ILP address before validating the path to avoid being tricked into participating in a Denial of Service (DoS) attack on a third-party endpoint.
+#### 4.3.2. Connection Migration (Receivers Only)
 
-#### 4.3.1. Connection Asset Details
+If an endpoint supports receiving, then the endpoint MAY change its ILP Address at any point during a connection by sending a `ConnectionNewAddress` frame. To ensure the new address is received and acknowledged, implementations MAY choose to send these frames only in ILP Prepare packets, although certain connections may not support this (e.g., a receiver emitting this frame to a non-receiving sender will only be able to propagate this frame in a fulfill or reject packet).
 
-Each endpoint MAY expose their asset details by sending a `ConnectionAssetDetails` frame. This frame is optional because some use-cases do not require it.
+Senders encountering this frame SHOULD wait for a separate, valid request/response (encrypted with the same shared secret) from the new address to validate the new path. STREAM relies upon this authenticated request/response packet flow in lieu of [QUIC's explicit Path Validation](https://quicwg.github.io/base-drafts/draft-ietf-quic-transport.html#rfc.section.6.7), so implementations SHOULD refrain from sending large numbers of packets or large amounts of data to a new ILP address before validating the path. For example, this might help avoid being tricked into participating in a Denial of Service (DoS) attack on a third-party endpoint.
 
-Asset details exposed by this frame MUST not change during the lifetime of a Connection.
+#### 4.3.3. Connection Asset Details
+Either endpoint MAY expose its asset details by sending a `ConnectionAssetDetails` frame in a stream packet.
+
+Asset details, whether exposed by this frame or obtained by a higher-layer protocol, MUST not change during the lifetime of a Connection. Therefore, if a receiver receives a `ConnectionAssetDetails` frame that contradicts existing asset details, then the receiver SHOULD close the connection because it would be ambiguous which asset details are authoritative.
```

Comment:
I don't think so - the current spec already [expressly prohibits](https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md#431-connection-asset-details) the connection details from changing, so this isn't changing that. However, with the new push to have STREAM endpoints get their `ConnectionAssetDetails` from a higher-layer protocol, that will increase the chances of divergent asset data (two sources instead of one). So this new verbiage just provides guidance around what should happen if one layer says something different from STREAM.

We could strike this guidance, but I think it's helpful as a `SHOULD` (but would be open to removing it too if you feel strongly).

Reply:
@kincaidoneil any thoughts here? If not feel free to resolve this.